src/backend/distributed/executor/local_executor.c

```diff
@@ -319,18 +319,20 @@ ExtractLocalAndRemoteTasks(bool readOnly, List *taskList, List **localTaskList,
 			 * nodes. We always prefer to use local placement, and require remote
 			 * placements only for modifications.
 			 */
-			task->partiallyLocalOrRemote = true;
 
-			Task *localTask = copyObject(task);
+			/* Don't push a task without any placements */
```

Comment:
This comment confuses me. Few lines ago we have `Assert(!(localTaskPlacementList == NIL && remoteTaskPlacementList == NIL));`, so we shouldn't have tasks without any placements here.

If we can have tasks without any placements, then why not change the assert with `if (localTaskPlacementList == NIL && remoteTaskPlacementList == NIL) continue;`?

Reply:
It's possible here for `remoteTaskPlacementList != NIL`

Reply:
0. What happens when a task doesn't have any local placements and is read only? Shouldn't we add it to *remoteTaskList?
1. The comment is still confusing. According to the assert, we cannot have "task without any placements", you might want to reword it to "task without any local placements"
2. The comment before this might need this updating. Intermediate results might not have both local and remote nodes.

Reply:
https://github.com/citusdata/citus/blob/master/src/backend/distributed/executor/adaptive_executor.c#L712

The current case is leaving localTaskList empty, so `tasksToExecute` comes out as it should be. It seems the comments are inaccurate, we claim `tasksToExecute` is the union of `localTaskList` & `remoteTaskList`, meanwhile this code path drops remote tasks